js/ci-configs/.gitlab-ci.yml

```
@@ -0,0 +1,79 @@
+image: node:8
+
+# Cache dependencies.
+cache:
+  paths:
+    - node_modules/
+
+stages:
+  - test
+  - lint
+  - deploy-gl-pages
+  - deploy-heroku
+
+# Define the desired stages.
+test:
+  stage: test
+  script:
+    - yarn install
+    - yarn test
+
+  # https://gitlab.com/gitlab-org/gitlab-ce/issues/29642
+  coverage: '/All files[^|]*\|[^|]*\s+([\d\.]+)/'
+
+lint:
+  stage: lint
+  script:
+    - yarn install
+    - yarn lint
+
+# Code quality setup.
+code_quality:
+  stage: test
```

Ah ok, this was also confusing for me that i only saw 2 green ticks instead of 3. This will fix it.

```suggestion
  stage: code_quality
```